tests/components/panasonic_viera/test_init.py

```diff
@@ -58,6 +100,37 @@ async def test_setup_entry_encrypted(hass):
         await hass.config_entries.async_setup(mock_entry.entry_id)
         await hass.async_block_till_done()
 
+        assert mock_entry.data[ATTR_DEVICE_INFO] == MOCK_DEVICE_INFO
+        assert mock_entry.unique_id == MOCK_DEVICE_INFO[ATTR_UDN]
+
+        state = hass.states.get("media_player.panasonic_viera_tv")
+
+        assert state
+        assert state.name == DEFAULT_NAME
+
+
+async def test_setup_entry_encrypted_missing_device_info_none(hass):
+    """Test setup with encrypted config entry and device info set to None."""
+    mock_entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id=MOCK_CONFIG_DATA[CONF_HOST],
+        data={**MOCK_CONFIG_DATA, **MOCK_ENCRYPTION_DATA},
+    )
+
+    mock_entry.add_to_hass(hass)
+
+    mock_remote = get_mock_remote(device_info=None)
+
+    with patch(
+        "homeassistant.components.panasonic_viera.Remote",
+        return_value=mock_remote,
+    ):
+        await hass.config_entries.async_setup(mock_entry.entry_id)
+        await hass.async_block_till_done()
+
+        assert mock_entry.data[ATTR_DEVICE_INFO] is None
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6319846)